### DISCUSSION TITLE
Helmfile configuration specs schema

### DISCUSSION THREAD

**Discussion by dreglad** [helmfile/helmfile#919] [Ideas]
I'd like to leverage IntelliSense tooling for improved developer experience when working with  helmfiles (i.e. autocompletion), which can be easily integrated by having some standard config schema definitions such as JSON Schema which I couldn't find being currently generated or distributed with Helmfile.

Do you think that there's a different approach that I overlooked for getting completion in my IDE for helmfile configuration?

In such case, would you consider that generating configuration schema manifest(s) is something in-scope to the project?

Would you accept a PR for generating such JSON Schemas?

**Comment by z0rc** [comment]
JSON schema won't work as LSP expects valid yaml files, while helmfile is go template of yaml, any template construct will break off-the-shelf yaml parser.
